src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1308,16 +1309,24 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
             // in order to avoid consuming resources in potentially many requests waiting for access.
             try
             {
-                if (_pool.EnableMultipleHttp2Connections)
+                if (!_concurrentStreams.TryRequestCreditNoWait(1))
                 {
-                    if (!_concurrentStreams.TryRequestCreditNoWait(1))
+                    if (_pool.EnableMultipleHttp2Connections)
```

Comment:
Was swapping these 2 if checks intentional?

Reply:
Yes. We only want to log `RequestLeftQueue` if we actually have to wait, so we have to perform the `TryRequestCreditNoWait` check first.
We already did so in the `EnableMultipleHttp2Connections == true`, so swapping the order around avoided duplicating a check.